mentat-vscode/src/webviews/components/ChatMessage.tsx

```diff
@@ -1,33 +1,164 @@
 import { VscAccount } from "react-icons/vsc";
-import { Message } from "types";
+import { FileEdit, Message, MessageContent } from "types";
 import MentatIcon from "./MentatIcon";
+import React from "react";
+import PillButton from "./PillButton";
 
-type Props = {
-    message: Message;
-};
-
+// TODO: Change the colors to vscode theme colors
 const light_theme: { [id: string]: string } = {
     prompt: "gray",
     code: "blue",
-    info: "cyan",
-    failure: "dark_red",
+    info: "lightblue",
+    failure: "darkred",
     success: "green",
-    input: "bright_blue",
+    input: "lightblue",
     error: "red",
     warning: "yellow",
 };
 const dark_theme: { [id: string]: string } = {
     prompt: "white",
     code: "blue",
-    info: "cyan",
-    failure: "bright_red",
+    info: "lightblue",
+    failure: "red",
     success: "green",
-    input: "bright_blue",
-    error: "red",
+    input: "lightblue",
+    error: "darkred",
     warning: "yellow",
 };
 
-// TODO: Once everything is working, make sure to memoize!!!
+function ContentPiece({ contentPiece }: { contentPiece: MessageContent }) {
```

Comment:
Some comments or docstrings for these would be helpful, or maybe a hyper-descriptive name